WIP: #16 authentication in api

bregySilvan opened the pull request:
## Description
- Add ```/login``` location for authentication. Currently we send back true or false. Later this will be an access token.
## Technical details
- location  ```/login```
- payload:  ``` { name: string, password: string} ```
- response:  ```{ success: boolean}```
- Used port can be changed in ```config.ts```

##  Start application
Let's use ```ts-node``` instead of ```tsc``` command. It compiles and executes ```.ts``` files directly. 
- make sure you have installed ```ts-node``` => ```npm install ts-node -g```
- to run server move into ```app/server``` and run ```npm start```
- to run client move into ```app/client``` and run ```npm start```
- do not forget to run ```npm install``` before first run, in both, server and client directory

## Note
In general we should avoid hardcoding stuff directly in classes. Pls put all variable data into ```config.ts``` file. Locations should also be added there.

bregySilvan committed:
#16 setup environment

-add app/client location
-add app/server location
-add npm script commands for running server & client

bregySilvan committed:
#16 add parameters

-add payload {name: string, password: string} and return type { success: boolean }
-server can be started via "npm run server"
-The Login api can be tested via url:
http://ip:port/login?name=silvan&password=bregy

bregySilvan committed:
#16 add login

-Add client side login request
-move package.json && tsconfig.json to root directory.

bregySilvan committed:
#16 fix login

-Fix client for /login location
-Move direct request response to client

bregySilvan committed:
#16 add ip detection

- Client now takes local ip address to connect

Christian-Seematter commented:
Nice working example of client-server communication. Decent info in the PR as well.

Will superagent be used within the angular app as well?

bregySilvan committed:
Merge branch 'master' into #16-authentication-in-API

Christian-Seematter reviewed (approved):
Source looks good, Server will be cleaned up by replacing it with the new version from #14. 

Ran and tested the connection with localhost, all good.

Christian-Seematter commented on package.json:
```diff
@@ -0,0 +1,24 @@
+  "scripts": {
+    "test": "gulp",
+    "start": "npm rn server",
+    "server": "cd app/server && tsc main.ts --outDir dist --lib ES2015 && cd dist/server && node main.js",
```
In the refactor, consider using tsc itself and configure the rest in tsconfig.json. With complexity of the app only compiling from the main.ts onwards might not be a clean solution anymore.

Same for the client instruction.

Christian-Seematter committed:
#16. Fix package-lock

npm would crash because of the strings in the package-lock. Assuming
this was a merge error.

bregySilvan committed:
#16 separate server and client

-Since the client and server are running independently they will be treated as separate programs. I created 2 separate package.json.
-Let's use "ts-node" to run our app.  It compiles file directly into RAM.

bregySilvan commented:
I seperated server and client to different apps because it doesn't make sense to open up a new issue for that.
In addition to that ```ts-node``` command got added for executing the application.

bregySilvan committed:
#add typings

-add typings
-little cleanup

bregySilvan commented:
@Christian-Seematter  
I looked up some tools for making requests and superagent seemed to be the smartest and most flexible one to use.

olinator98 commented:
@bregySilvan The client is now ready for the integration of the authentication. 
Cherio

bregySilvan commented:
@olinator98 
Go ahead and create the PR for your client which is finished. We can merge your changes on the client first and afterward I can commit my login integration directly into #18.

bregySilvan committed:
Merge branch 'master' into #16-authentication-in-API
